From 386f094c9fc7937a7d8624a3a015d213bade6875 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:50:41 -0400
Subject: [fs] nfsd: fix write reply size estimate

Message-id: <1403560314-32525-100-git-send-email-bfields@redhat.com>
Patchwork-id: 84498
O-Subject: [Patch RHEL7 099/172] nfsd4: fix write reply size estimate
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

The write reply also includes count and stable_how.

Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: f34e432b673781175c782fa8c5a619114340ba38
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4proc.c b/fs/nfsd/nfs4proc.c
index 31ce60f..b4057fc 100644
--- a/fs/nfsd/nfs4proc.c
+++ b/fs/nfsd/nfs4proc.c
@@ -1527,7 +1527,7 @@ static inline u32 nfsd4_setclientid_rsize(struct svc_rqst *rqstp, struct nfsd4_o
 
 static inline u32 nfsd4_write_rsize(struct svc_rqst *rqstp, struct nfsd4_op *op)
 {
- return (op_encode_hdr_size + op_encode_verifier_maxsz) * sizeof(__be32);
+ return (op_encode_hdr_size + 2 + op_encode_verifier_maxsz) * sizeof(__be32);
 }
 
 static inline u32 nfsd4_exchange_id_rsize(struct svc_rqst *rqstp, struct nfsd4_op *op)
-- 
1.7.1